src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -1022,10 +1022,18 @@ public class TraceStatus
             /// <value></value>
             public string? Hostname
             {
-                get => _status.Destination != IPAddress.Any.ToString()
-                    && _status.Destination != IPAddress.IPv6Any.ToString()
-                        ? _status.Destination
-                        : null;
+                get
+                {
+                    if (_status.Address == IPAddress.Any
+                        || _status.Address == IPAddress.IPv6Any)
+                    {
+                        // There was no response to the ping (TimedOut).
+                        return null;
+                    }
+
+                    // We have a usable IP address from this ping reply.
+                    return _status.Destination;
```

Comment:
We initialize TraceStatus with resolvedTargetName coming from InitProcessPing(). So maybe we should do the check and assign to null in InitProcessPing() ?

> // There was no response to the ping (TimedOut).

I see we use `targetAddress` to initialize TraceStatus but the value is not related to ping status and timeout.

Reply:
Not 100% clear on what you're suggesting here. Just move this check to the constructor and set the value there?

Yes, TraceStatus isn't directly tied to PingStatus, but it does effectively "wrap" the PingStatus and surfaces some of the properties from the PingStatus while adding some extra properties to make the traceroute progress clear.

Reply:
Moved this check to the constructor so we only need to check it once. 👍